src/Form/Form.js

```diff
@@ -60,14 +63,26 @@ class Form {
         if (this.handlerExecuted) return
 
         const credentials = this.getValues()
-        if (credentials.password) {
-            // ask to store credentials and/or fireproof
-            if (this.shouldPromptToStoreCredentials) {
-                // @ts-ignore
-                this.device.storeCredentials(credentials)
-            }
-            this.handlerExecuted = true
+
+        // do nothing if password was absent
+        if (!credentials.password) return
+
+        // checks to determine if we should offer to store credentials and/or fireproof
+        const checks = [
+            this.shouldPromptToStoreCredentials,
+            this.device.shouldPromptToStoreCredentials({
+                formElement: this.form
+            })
+        ]
+
+        // if *any* of the checks are truthy, proceed to offer
+        if (checks.some(Boolean)) {
+            this.device.storeCredentials(credentials)
         }
+
+        // mark this form as being handled
+        // TODO(Shane): is this correct, what happens if a failed submission is retried?
+        this.handlerExecuted = true
```

Comment:
If the user changes the form values, this is set again to `false`. There is a project to improve on this stuff, but it's out of scope here. There are lots of implications to consider.